x-pack/legacy/plugins/apm/typings/elasticsearch/aggregations.ts

```diff
@@ -0,0 +1,239 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Unionize } from 'utility-types';
+
+type SortOrder = 'asc' | 'desc';
+type SortInstruction = Record<string, SortOrder | { order: SortOrder }>;
+export type SortOptions = SortOrder | SortInstruction | SortInstruction[];
+
+type Script =
+  | string
+  | {
+      lang?: string;
+      id?: string;
+      source?: string;
+      params?: Record<string, string | number>;
+    };
+
+type BucketsPath = string | Record<string, string>;
+
+type SourceOptions = string | string[];
+
+type MetricsAggregationOptions =
+  | {
+      field: string;
+      missing?: number;
+    }
+  | {
+      script?: Script;
+    };
+
+interface MetricsAggregationResponsePart {
+  value: number | null;
+}
+
+export interface AggregationOptionsByType {
+  terms: {
+    field: string;
+    size?: number;
+    missing?: string;
+    order?: SortOptions;
+  };
+  date_histogram: {
+    field: string;
+    format?: string;
+    min_doc_count?: number;
+    extended_bounds?: {
+      min: number;
+      max: number;
+    };
+  } & ({ calendar_interval: string } | { fixed_interval: string });
+  histogram: {
+    field: string;
+    interval: number;
+    min_doc_count?: number;
+    extended_bounds?: {
+      min: number;
+      max: number;
+    };
+  };
+  avg: MetricsAggregationOptions;
+  max: MetricsAggregationOptions;
+  min: MetricsAggregationOptions;
+  sum: MetricsAggregationOptions;
+  value_count: MetricsAggregationOptions;
+  cardinality: MetricsAggregationOptions & {
+    precision_treshold?: number;
+  };
+  percentiles: {
+    field: string;
+    percents?: number[];
+  };
+  extended_stats: {
+    field: string;
+  };
+  top_hits: {
+    from?: number;
+    size?: number;
+    sort?: SortOptions;
+    _source?: SourceOptions;
+  };
+  filter: Record<string, any>;
+  filters: {
+    filters: Record<string, any> | any[];
```

Comment:
As long as we're adding such strong request types, it bothers me that the types for filters here are `any`. You've already defined the `ESFilter` type below, and there is probably an extension for `NamedEsFilter` that can be used here